astropy/units/scripts/typestubs_for_units.py

```diff
@@ -0,0 +1,77 @@
+"""
+Simple stub generator. Focused on `astropy.units` for now.
+
+Stub files (.pyi) help type checkers understand dynamically created code.
+
+TO DO: functions, functools._lru_cache_wrapper, method, DataInfoMeta.
+"""
+
+import astropy.units as units
+
+
+def main():
+    with open("astropy/units/__init__.pyi", "w") as f:
+        f.write("# This stub file was automatically generated.\n")
+        for line in generate_stub_lines():
+            f.write("\n")
+            f.write(line)
+
+
+def generate_stub_lines():
+    # --- preprocessing ---
+    import_lines = []
+    variable_lines = []
+    for name in units.__all__:
+        obj = getattr(units, name)
+        # instances of unit classes
+        if isinstance(obj, (units.UnitBase, units.FunctionUnitBase)):
+            type_name = type(obj).__name__
+            docstring = f'"""{obj.__doc__}"""\n' if has_own_docstring(obj) else ""
+            variable_lines.append(f"{name}: {type_name}\n{docstring}")
+            # if the object type is not part of astropy.units, we need to import it
+            if not getattr(units, type(obj).__name__, None):
+                import_lines.append(make_import_line(type(obj)))
+        # classes
+        elif type(obj).__name__ in ("type", "ABCMeta", "_UnitMetaClass"):
+            # "as"-suffix needed to make type checkers find it
+            import_lines.append(make_import_line(obj, include_as_suffix=True))
+
```

Comment:
Would it make sense to have a kind of debug option so that one could find which entries fail? One option might be to raise a normally invisible warning here, like `else: warn(...)`.

Reply:
@mhvk sure! why not. Right now, as indicated in the top of the script, the following types will fail: functions, functools._lru_cache_wrapper, method, DataInfoMeta. I skipped those to save developer time, since they are a tinsy fraction. I can do a `logging.info` akin to other scripts.

Reply:
Those shouldn't be in `Astropy.units.__all__`! So definitely good not to include them.

Reply:
@nstarman I'm glad my gut instinct here was similar to yours. :)